open_samus_returns_rando/patcher_editor.py

```diff
@@ -56,3 +56,10 @@ def copy_actor(self, scenario: str, coords, templateActor: Container, newName: s
         newActor.z = coords[2] + offset[2]
 
         return newActor
+    
+    def remove_entity(self, reference: dict):
+        scenario = self.get_scenario(reference["scenario"])
+        layer = reference.get("layer", "default")
```

Comment:
Either make sure that layer always exists in the reference or use `0` as default.
The layers in SR have no names, they are indeed based. That's a copy of dread but it doesn't take the difference into account :p

Reply:
TL;DR: `layer = reference["layer"]`

On a second thought: I would prefer to not use any default here.
If we want to remove an entity, we have to specifiy on which layer it is.

Layer 0 contains the triggers and so it will fail anyways for all non-trigger entities we want to remove, if they haven't specified the layer.